connector/src/main/java/org/geysermc/connector/network/session/GeyserSession.java

```diff
@@ -912,8 +912,14 @@ protected void tick() {
             lastMovementTimestamp = System.currentTimeMillis();
         }
 
-        for (Tickable entity : entityCache.getTickableEntities()) {
-            entity.tick(this);
+        try {
+            synchronized (entityCache.getTickableEntities()) {
+                for (Tickable entity : entityCache.getTickableEntities()) {
+                    entity.tick(this);
+                }
+            }
+        } catch (Exception e) {
```

Comment:
Is this leftover debug code?

Reply:
Yeah, apparently exceptions from the tick thread are not printed out.

Reply:
I feel like exceptions should be printed if debug mode is enabled though. May be worth surrounding the GeyserSession tick method with a try/catch where that’s called instead, and print the exceptions if debug mode is set.

Reply:
I personally don't think it should be limited to debug mode. Nothing should throw an exception anywhere in the code except if something is wrong, and we want to know when that happens.